Makefile

```
@@ -388,21 +388,28 @@ docs/%: export TFDOCS_RECURSIVE ?= $(if $(wildcard $(TFDOCS_PATH)/modules),--rec
 docs/%: export TFDOCS_TEMPLATE ?= <!-- BEGIN TFDOCS -->\n{{ .Content }}\n\n<!-- END TFDOCS -->
 docs/%: export TFDOCS_FILE ?= README.md
 docs/%: export TFDOCS_PATH ?= .
-docs/%: export README_FILES ?=  $(if $(wildcard $(TFDOCS_FILE)),true,)
-docs/%: export TF_FILES ?= $(if $(wildcard $(TFDOCS_PATH)/*.tf),true,)
+docs/%: export README_FILES ?=  $(if $(wildcard $(TFDOCS_FILE)),true,$(if $(wildcard modules/*/$(TFDOCS_FILE)),true,))
+docs/%: export TF_FILES ?= $(if $(wildcard $(TFDOCS_PATH)/*.tf),true,$(if $(wildcard modules/*/*.tf),true,))
 docs/%: export HCL_FILES ?= $(if $(wildcard $(TFDOCS_PATH)/*.pkr.hcl),true,)
 docs/%: export TFCMD_OPTS ?= $(if $(README_FILES),$(if $(TF_FILES),$(TFDOCS_MODULES_OPTIONS),$(if $(HCL_FILES),$(HCLDOCS_MODULES_OPTIONS),)),)
 docs/%: export TFDOCS_CMD ?= $(if $(TFCMD_OPTS),$(TFDOCS) $(TFCMD_OPTS) $(TFDOCS_OPTIONS),)
 docs/%: export TFDOCS_LINT_CMD ?=  $(if $(TFCMD_OPTS),$(TFDOCS) --output-check $(TFCMD_OPTS) $(TFDOCS_OPTIONS),)
+docs/%: export MARKDOWN_FILES ?= $(shell $(GIT_LS_FILES) -- '*.md' $(FIND_EXCLUDES))
 
 ## Generates Terraform documentation
 docs/generate: | terraform/format
 	@[ "${TFDOCS_CMD}" ] && ( echo "[$@]: Generating docs";) || ( echo "[$@]: No docs to generate";)
 	$(TFDOCS_CMD)
-	@if [ -n $(README_FILES) ] && [ "$$(tail -c1 $(TFDOCS_FILE))" != "$('\n')" ]; then \
-		echo "Adding newline to the end of $(TFDOCS_FILE) file"; \
-		echo "" >> $(TFDOCS_FILE); \
-	fi
+	@make docs/format
+
+## Adds a newline to the end of all project markdown files if one does not already existGenerates Terraform documentation
+docs/format:
+	@for markdown_file in $(MARKDOWN_FILES); do \
+		if [ "$$(tail -c1 $$markdown_file)" != "$('\n')" ]; then \
+			echo "Adding newline to the end of $$markdown_file file"; \
```

The double $ should only be needed when using $(...) or ${...}. Can you confirm the output is what you expect?

```suggestion
			echo "Adding newline to the end of $${markdown_file} file"; \
```

---

Here's the output of a test run with three files to update without the suggested changes:
```
README.md updated successfully
modules/test/README.md updated successfully
modules/test2/README.md updated successfully
Adding newline to the end of modules/test/README.md file
Adding newline to the end of modules/test2/README.md file
Adding newline to the end of README.md file
```
I get the same exact output after making the suggested changes.  Makefiles are still not my friend, so I'm not sure why both versions work the same.  I went ahead and left the changes in.